core/__tests__/AsyncStorage.test.ts

```diff
@@ -0,0 +1,160 @@
+import AsyncStorage from '../src/AsyncStorage';
+
+class StorageMock implements IStorageBackend<any> {
+  getSingle = jest.fn();
+  setSingle = jest.fn();
+  getMany = jest.fn();
+  setMany = jest.fn();
+  removeSingle = jest.fn();
+  removeMany = jest.fn();
+  getKeys = jest.fn();
+  dropStorage = jest.fn();
+}
+
+describe('AsyncStorage', () => {
+  const mockedStorage = new StorageMock();
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+  });
+
+  describe('main API', () => {
+    it.each(['get', 'set', 'remove'])(
+      'handles single %s api call',
+      async (methodName: string) => {
+        const as = new AsyncStorage(mockedStorage, {
+          logger: false,
+          errorHandler: false,
+        });
+
+        const key = 'myKey';
+        const value = {
+          name: 'Jerry',
+        };
+
+        switch (methodName) {
+          case 'get': {
```

Comment:
Yikes, that's not what I meant, this code is almost no different. You can use each like this:
```js
it.each([
  ['get', 'myKey'], 
  ['set', 'myKey', {name: 'Jerry'}], 
  ['remove', 'myKey']
])('handles single %s api call', async (methodName: 'get' | 'set' | 'remove', ...args) => {
  await as[methodName](...args);
  expect(mockedStorage.getSingle).toBeCalledWith(...args.concat(null));
}
```

Reply:
yh, shame. Will cover this in next PR.